apps/src/templates/DataWorkspace.jsx

```diff
@@ -1,16 +1,157 @@
+  },
+  render() {
+    return <a style={styles.link} href='#'>{this.props.name}</a>;
+  }
+});
+
+const EditTableRow = React.createClass({
+  propTypes: {
+    tableName: React.PropTypes.string.isRequired
+  },
+  render() {
+    return (
+      <tr style={styles.editRow}>
+        <td style={styles.cell}>
+          <EditLink name={this.props.tableName}/>
+        </td>
+        <td style={styles.cell}>
+          <button className='btn btn-danger' style={styles.button}>Delete</button>
+        </td>
+      </tr>
+    );
+  }
+});
+
+const AddTableRow = React.createClass({
+  render() {
+    return (
+      <tr style={styles.addRow}>
+        <td style={styles.cell}>
+          <input style={styles.input} placeholder={msg.dataTableNamePlaceholder()}></input>
+        </td>
+        <td style={styles.cell}>
+          <button className='btn btn-primary' style={styles.button}>Add</button>
+        </td>
+      </tr>
+    );
+  }
+});
 
 const DataWorkspace = React.createClass({
   propTypes: {
     style: React.PropTypes.object.isRequired
   },
-
   render() {
     return (
-      <div id="dataWorkspaceWrapper" style={this.props.style}>
-        Data workspace
+      <div id='dataWorkspaceWrapper' style={this.props.style}>
+        <PaneHeader
+            id='headers'
+            dir={this.props.localeDirection}
+            hasFocus={!this.props.isRunning}
+            className={this.props.isRunning ? 'is-running' : ''}>
+          <div id='dataModeHeaders'>
+            <PaneSection id='workspace-header'>
+              <span id='workspace-header-span'>
+                {msg.dataWorkspaceHeader()}
+              </span>
+            </PaneSection>
+          </div>
+        </PaneHeader>
+
+        <div id='data-mode-container' style={styles.container}>
+          <div id='data-table-container' style={styles.tableContainer}>
+            <h4>Data</h4>
+
+            <table style={styles.table}>
+              <tbody>
+                <tr style={styles.editRow}>
+                  <td style={styles.cell}>
+                    <EditLink name={msg.keyValuePairLink()}/>
+                  </td>
+                </tr>
+              </tbody>
+            </table>
+
+            <table style={styles.table}>
+              <colgroup>
+                <col width={tableWidth - buttonColumnWidth}/>
+                <col width={buttonColumnWidth}/>
+              </colgroup>
+              <tbody>
+                {/* placeholder table names, to be populated from Firebase */}
+                <EditTableRow tableName="Table 1"/>
+                <EditTableRow tableName="Table 2"/>
+                <AddTableRow/>
+              </tbody>
+           </table>
+          </div>
+        </div>
+>>>>>>> add initial content to dataWorksapce
       </div>
     );
   }
 });
-export default DataWorkspace;
+
+export default connect(state => ({
+  localeDirection: state.pageConstants.localeDirection,
+  isRunning: !!state.runState.isRunning
```

Comment:
Oooh, that would be sweet. The other things I'd been considering trying to build (if such a thing doesn't exist already) is the ability to auto-populate propTypes based on info provided to connect, so that I don't end up having to list all these props in two places.